wa/workloads/aitutu/uiauto/app/src/main/java/com/arm/wa/uiauto/aitutu/UiAutomation.java

```diff
@@ -0,0 +1,140 @@
+/*    Copyright 2013-2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+
+package com.arm.wa.uiauto.aitutu;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.graphics.Rect;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.test.uiautomator.UiObject;
+import android.support.test.uiautomator.UiObjectNotFoundException;
+import android.support.test.uiautomator.UiSelector;
+import android.support.test.uiautomator.UiScrollable;
+import android.view.KeyEvent;
+import android.util.Log;
+
+import com.arm.wa.uiauto.BaseUiAutomation;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.TimeUnit;
+
+
+@RunWith(AndroidJUnit4.class)
+public class UiAutomation extends BaseUiAutomation {
+
+    public static String TAG = "UXPERF";
+
+    @Test
+    public void setup() throws Exception {
+        clearPopups();
+        downloadAssets();
+    }
+
+    @Test
+    public void runWorkload() throws Exception {
+        runBenchmark();
+    }
+
+    @Test
+    public void extractResults() throws Exception {
+        getScores();
+    }
+
+    public void clearPopups() throws Exception {
+        UiSelector selector = new UiSelector();
+
+        UiObject cancel = mDevice.findObject(selector.textContains("CANCEL")
+                                             .className("android.widget.Button"));
+        waitObject(cancel);
+        cancel.click();
```

Comment:
This should have a timeout and conditionally click on cancel if the popup is present.